edweiss-app/app/(app)/deck/[id]/card/index.tsx

```diff
@@ -0,0 +1,218 @@
+/**
+ * Create a card or edit an existing card screen
+ * 
+ * @file index.tsx
+ * @description Create a card or edit an existing card screen
+ * @author Tuan Dang Nguyen
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import RouteHeader from '@/components/core/header/RouteHeader';
+import { ApplicationRoute } from '@/constants/Component';
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import Icon from '@/components/core/Icon';
+import RichText from '@/components/core/rich-text/RichText';
+import TText from '@/components/core/TText';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { callFunction } from '@/config/firebase';
+import { iconSizes } from '@/constants/Sizes';
+import { useRepositoryDocument } from '@/hooks/repository';
+import { useStringParameters } from '@/hooks/routeParameters';
+import Memento from '@/model/memento';
+import { checkDupplication_EmptyField } from '@/utils/memento/utilsFunctions';
+import { router } from 'expo-router';
+import React, { useState } from 'react';
+import { Modal } from 'react-native';
+import { DecksRepository } from '../../_layout';
+
+// ------------------------------------------------------------
+// ----------------- CreateCardScreen Component ----------------
+// ------------------------------------------------------------
+
+/**
+ * Create or edit card screen
+ * User can create a new card or edit an existing card with a question and an answer
+ * 
+ * @returns {ApplicationRoute} Screen to create or edit a card
+ */
+const CreateEditCardScreen: ApplicationRoute = () => {
+    const { deckId, mode, prev_question, prev_answer, cardIndex } = useStringParameters();
```

Comment:
As this route takes quite a lot of parameters, I suggest you look at `utils/routeParameters.ts` to make this route type safe and future-proof for further improvements and expansions.

Reply:
Yup, I will take a look at that. I have just also learned from Florian that this exists.